src/cliffordsynthesis/encoding/STQGatesEncoder.cpp

```diff
@@ -0,0 +1,236 @@
+//
+// Created by Velsh Aleksei on 16.06.23.
+//
+
+#include "cliffordsynthesis/encoding/STQGatesEncoder.hpp"
+
+#include "LogicTerm/LogicTerm.hpp"
+#include "utils/logging.hpp"
+
+namespace cs::encoding {
+
+using namespace logicbase;
+
+void encoding::STQGatesEncoder::createSingleQubitGateVariables() {
+  DEBUG() << "Creating single-qubit gate variables for the STQEncoding.";
+  vars.gS.reserve(T);
+  for (std::size_t t = 0U; t < T; ++t) {
+    auto& timeStep = vars.gS.emplace_back();
+    timeStep.reserve(SINGLE_QUBIT_GATES_FOR_STQ_ENCODING.size());
+    for (const auto gate : SINGLE_QUBIT_GATES_FOR_STQ_ENCODING) {
+      auto& g = timeStep.emplace_back();
+      g.reserve(N);
+      for (std::size_t q = 0U; q < N; ++q) {
+        const std::string gName = "g_" + std::to_string(t) + "_" +
+                                  toString(gate) + "_" + std::to_string(q);
+        TRACE() << "Creating variable " << gName;
+        g.emplace_back(lb->makeVariable(gName));
+      }
+    }
+  }
+}
+
+void encoding::STQGatesEncoder::createTwoQubitGateVariables() {
```

Comment:
As it stands, much code is repeated from `GateEncoder.cpp`. You could call the base class method and then loop over the timesteps to insert the identity variables separately. Or don't overwrite the method and use `g_i_i` as the variable for the identity gate.